src/main/kotlin/xyz/crossward/controller/StatsController.kt

```diff
@@ -0,0 +1,39 @@
+package xyz.crossward.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.bind.annotation.*
+import xyz.crossward.service.StatsService
+
+@RestController
+@RequestMapping("/api/stats")
+class StatsController(
+    var service: StatsService
+) {
+    @GetMapping("/average")
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional(readOnly = true)
+    fun getAverageTimeByUserId(
+        @RequestParam("user", required = false) user: String?
```

Comment:
Let's name the query and method parameters to `user_id` and `userId`  to make this consistent with the user controller.